server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -528,6 +529,11 @@ public void getShuffleResult(
     String appId = request.getAppId();
     int shuffleId = request.getShuffleId();
     int partitionId = request.getPartitionId();
+    BlockIdLayout blockIdLayout =
```

Comment:
Could you consider some compatible issues? If older client connects a new server, the blockIdLayout may be null.

Reply:
We could fall back to `BlockIdLayout.DEFAULT' here, but that would only work for clients that use the original values in `Constants.java`. For cluster setups where `Constants.java` has been changed, there is no way to pick the right block id layout here.

Reply:
Yes, what do you suggest? Use `BlockIdLayout.DEFAULT` if `request.getBlockIdLayout == null` and mark this a breaking change for users that deployed a cluster with modified bit lengths in `Constants.java`? How can we make this compatible for those users?

Reply:
We can use `BlockIdLayout.DEFAULT` if layout is null. We can't guarantee the compatibility if users modify our code in the project.

Reply:
I like this, done!